src/library-authoring/data/apiHook.ts

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { useQuery } from '@tanstack/react-query';
+import { MeiliSearch } from 'meilisearch';
+
+import { useContentSearchConnection, useContentSearchResults } from '../../search-modal';
+import { getContentLibrary } from './api';
+
+/**
+ * Hook to fetch a content library by its ID.
+ */
+export const useContentLibrary = (libraryId?: string) => (
+  useQuery({
+    queryKey: ['contentLibrary', libraryId],
+    queryFn: () => getContentLibrary(libraryId),
+  })
+);
+
+/**
+ * Hook to fetch the count of components and collections in a library.
+ */
+export const useLibraryComponentCount = (libraryId: string, searchKeywords: string) => {
```

Comment:
This is temporary right? In the future we should have a `<SearchContextProvider>` on the Library Home page, and there should be no need to initialize a Meilisearch client manually.

Reply:
Yes! This was refactored here: https://github.com/openedx/frontend-app-course-authoring/pull/1141